primitives/consensus/slots/src/lib.rs

```diff
@@ -21,8 +21,38 @@
 
 use codec::{Decode, Encode};
 
-/// A slot number.
-pub type SlotNumber = u64;
+/// Unit type wrapper that represents a slot.
+#[derive(Debug, Encode, Decode, PartialEq, Eq, PartialOrd, Clone, Copy, Default, Ord)]
+pub struct Slot(pub u64);
```

Comment:
What do you think about not exposing the inner value and just using `Deref` instead? For some reason I find code like `slot.0` really ugly 🙈